ffi/diplomat/src/pluralrules.rs

```diff
@@ -132,6 +134,10 @@ pub mod ffi {
         /// Construct for a given string representing a number
         #[diplomat::rust_link(icu::plurals::PluralOperands::from_str, FnInStruct)]
         pub fn create(s: &str) -> DiplomatResult<ICU4XPluralOperands, ICU4XError> {
+            // #2520
+            if str::from_utf8(s.as_bytes()).is_err() {
+                return Err(ICU4XError::PluralParserError).into();
+            }
```

Comment:
Thought/Suggestion: The other way to do this would be to wire this function to use the FixedDecimal code path.